index.js

```diff
@@ -71,7 +83,10 @@ const main = async () => {
     core.setOutput("task-arn", taskArn);
 
     core.debug("Waiting for task to finish...");
-    await ecs.waitFor("tasksStopped", { cluster, tasks: [taskArn] }).promise();
+    await retry_promise(
+      3,
+      () => ecs.waitFor("tasksStopped", { cluster, tasks: [taskArn] }).promise(),
+    )
```

Comment:
did you check the documentation? There is something called as https://github.com/aws/aws-sdk-js/blob/307e82673b48577fce4389e4ce03f95064e8fe0d/lib/service.d.ts#L53
there is maybe a high chance, that you could pass `$waiter` there and just extend those limits... I will be expecting that this function will have some similar arguments.

Reply:
https://github.com/aws/aws-sdk-js/blob/cf8fe5adad400bb6a918843aed0f80ee47fe3a48/clients/ecs.d.ts#L434 this exactly

Reply:
I had checked the docs (not the source), and I didn't see this there. Great catch. 👌

Looking at https://github.com/aws/aws-sdk-js/blob/cf8fe5adad400bb6a918843aed0f80ee47fe3a48/lib/service.d.ts#L6, I agree we should just raise `maxAttempts`. Digging for the current value.

Reply:
Current [is 100](https://github.com/aws/aws-sdk-js/blob/2e5ecc6fe766f0aff1c8dd482871e626cf451869/apis/ecs-2014-11-13.waiters2.json#L32). Making the PR go for 300.